### PR TITLE
VPN-5443: Notify accessibility clients of alerts

### PR DESCRIPTION
**Pull request by vinocher**
## Description

   Notify accessibility clients, like screen readers, of alert strings like "Incorrect password", "Signed out and device disconnected", so that the screen reader will read those. Also notify when connection state and health changes.
   This uses the QAccessible::ValueChanged and QAccessible::updateAccessibility, which has been verified to work on Windows. Unfortunately, Qt doesn't implement this fully on other platforms. Another ticket will be opened to track investigation of potential fixes on other platforms.

## Reference

[VPN-5443](https://mozilla-hub.atlassian.net/browse/VPN-5443), [GitHub 7863](https://github.com/mozilla-mobile/mozilla-vpn-client/issues/7863)

## Checklist
    
- [x] My code follows the style guidelines for this project
- [x] I have not added any packages that contain high risk or unknown licenses (GPL,  LGPL, MPL, etc. consult with DevOps if in question)
- [x] I have performed a self review of my own code
- [x] I have commented my code PARTICULARLY in hard to understand areas
- [ ] I have added thorough tests where needed

[VPN-5443]: https://mozilla-hub.atlassian.net/browse/VPN-5443?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/apps/vpn/ui/screens/home/controller/ConnectionStability.qml [hunk 1]
```diff
@@ -22,6 +22,19 @@ Item {
...
+
+    function handleConnectionHealthStateChange() {
+        // If the stability item is not visible, there are no connection health problems to be be processed
+        if (!visible) {
+            return;
+        }
+
+        // Notify accessibility client of connection health problems
+        let notificationText = stabilityLabel.text + " " + stabilityLabelInstruction.text;
+        console.log("MZAccessibleNotification " + notificationText);
```
Comment (MattLichtenstein) [inline src/apps/vpn/ui/screens/home/controller/ConnectionStability.qml:34]: If leaving the log messages in feels necessary, would it make more sense to log it in `AccessibleNotification::Notify`?

**Review by MattLichtenstein — COMMENTED** [review]
Wasn't able to fully test this on macOS, but by looking at the log messages, I noticed that with this change, we are producing quite a bit of redundancy in the number of accessible notifications when changing controller states. Examples:

App startup: 
```
Debug: MZAccessibleNotification   
Debug: MZAccessibleNotification VPN is off  
Debug: MZAccessibleNotification VPN is off Turn on to protect your privacy  
```

Turning the VPN on:
```
Debug: MZAccessibleNotification Connecting… Turn on to protect your privacy  
Debug: MZAccessibleNotification Connecting… Masking connection and location  
Debug: MZAccessibleNotification VPN is off Masking connection and location  
Debug: MZAccessibleNotification VPN is off Turn on to protect your privacy  
Debug: MZAccessibleNotification Connecting… Turn on to protect your privacy  
Debug: MZAccessibleNotification Connecting… Masking connection and location  
Debug: MZAccessibleNotification Connecting… Secure and private •   
Debug: MZAccessibleNotification Connecting… Secure and private •  
Debug: MZAccessibleNotification VPN is on Secure and private •  
```

Turning the VPN off:
```
Debug: MZAccessibleNotification Disconnecting… Secure and private •  
Debug: MZAccessibleNotification Disconnecting… Unmasking connection and location Secure and private • 
Debug: MZAccessibleNotification Disconnecting… Unmasking connection and location  
Debug: MZAccessibleNotification VPN is on Unmasking connection and location  
Debug: MZAccessibleNotification VPN is on  
Debug: MZAccessibleNotification VPN is on Secure and private •   
Debug: MZAccessibleNotification VPN is on   
Debug: MZAccessibleNotification VPN is off   
Debug: MZAccessibleNotification VPN is off Turn on to protect your privacy 
```

Losing signal: 
```
Debug: MZAccessibleNotification VPN is on  
Debug: MZAccessibleNotification No Signal Check Connection 
Debug: MZAccessibleNotification No Signal Check Connection 
```

Regaining signal:
```
Debug: MZAccessibleNotification Connecting…  
Debug: MZAccessibleNotification Connecting… Masking 
Debug: MZAccessibleNotification Unstable Check Connection 
Debug: MZAccessibleNotification Connecting… Secure and private •  
Debug: MZAccessibleNotification Connecting… Secure and private •  
Debug: MZAccessibleNotification VPN is on Secure and private •   
```

Of course, this PR doesn't control the different UI states the VPN goes through, but do you think this will create too much extra noise for a screen reader? Do we need to clean this up (in a separate ticket of course) before adding accessibility here?

File: src/apps/vpn/ui/screens/home/controller/ConnectionStability.qml (hunk 1 above)
Comment (vinocher) [reply src/apps/vpn/ui/screens/home/controller/ConnectionStability.qml]: Good catch! Fixed.

**Comment by vinocher** [comment]
> Wasn't able to fully test this on macOS, but by looking at the log messages, I noticed that with this change, we are producing quite a bit of redundancy in the number of accessible notifications when changing controller states.

Yes, this is because the UI goes through may intermediate state changes which are too quick for the sighted user to notice. The corresponding Accessibility notifications received by the screen reader are interrupted when new ones are received. In most cases the new ones are received before the old ones have been queued for speech rendering, so the old ones are dropped by the screen reader, and the user experience is fine.

**Review by MattLichtenstein — APPROVED** [review]

**Pull request merged**